dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentTypeLoader.java

```diff
@@ -0,0 +1,303 @@
+                    .append(field.variable())
+                    .append("DBFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+            } else if (field instanceof TimeField) {
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("ShortFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append(" =\"[ ")
+                    .append(field.variable())
+                    .append("\" ])");
+            } else if (field instanceof DateTimeField) {
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append(" =\"[ ")
+                    .append(field.variable())
+                    .append(" ] \")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("ShortFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("LongFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+                sb.append("#set( $")
+                    .append(field.variable())
+                    .append("DBFormat =\"[ ")
+                    .append(field.variable())
+                    .append(" ]\")");
+            }
+        }
+
+        // sets the categories as a list on velocity
+        sb.append("#set( $ContentletCategories =[] )");
+        sb.append("#set( $ContentletCategoryNames =[] )");
+
+
+        // This is code is repeated becuase the bug GETTYS-268, the content variables were been
+        // overwritten
+        // by the parse inside the some of the content fields
+        // To edit the look, see WEB-INF/velocity/static/preview/content_controls.vtl
+
+        sb.append("#set( $EDIT_CONTENT_PERMISSION =$EDIT_CONTENT_PERMISSION")
+            .append(fakeIdentifier)
+            .append(" )");
+        sb.append("#set( $CONTENT_INODE ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $IDENTIFIER_INODE ='")
+            .append(fakeIdentifier)
+            .append("' )");
+
+        sb.append("#set( $ContentInode ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $ContentIdentifier ='")
+            .append(fakeIdentifier)
+            .append("' )");
+        sb.append("#set( $ContentletTitle =\"")
+            .append(UtilMethods.espaceForVelocity(fakeTitle))
+            .append("\" )");
+        sb.append("#set( $isWidget = false)");
+
+        return writeOutVelocity(filePath, sb.toString());
+    }
+
+    public void removeContentTypeFile(ContentType contentType) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)